deltacat/compute/compactor/model/compact_partition_params.py

```diff
@@ -414,13 +433,69 @@ def num_rounds(self, num_rounds: int) -> None:
         self["num_rounds"] = num_rounds
 
     @property
-    def parquet_to_pyarrow_inflation(self) -> int:
+    def parquet_to_pyarrow_inflation(self) -> float:
+        """
+        The inflation factor for the parquet uncompressed_size_bytes to pyarrow table size.
+        """
         return self["parquet_to_pyarrow_inflation"]
 
     @parquet_to_pyarrow_inflation.setter
-    def parquet_to_pyarrow_inflation(self, value: int) -> None:
+    def parquet_to_pyarrow_inflation(self, value: float) -> None:
         self["parquet_to_pyarrow_inflation"] = value
 
+    @property
+    def enable_input_split(self) -> bool:
+        """
+        When this is True, the input split will be always enabled for parquet files.
+        The input split feature will split the parquet files into individual row groups
+        so that we could process them in different nodes in parallel.
+        By default, input split is enabled for incremental compaction and disabled for rebase.
+        """
+        return self["enable_input_split"]
+
+    @enable_input_split.setter
+    def enable_input_split(self, value: bool) -> None:
+        self["enable_input_split"] = value
+
+    @property
+    def force_using_previous_inflation_for_memory_calculation(self) -> bool:
```

Comment:
Wouldn't an enum here be more extensible than a flag boolean arguemnt? The driver could pass a strategy for memory calculation - like `MemoryCalculation.DEFAULT, MemoryCalculation.USE_PREVIOUS_INFLATION, MemoryCalculation.XXX`

Reply:
Good call out. I will address this soon.